### PR TITLE
[getopt-win32] fix static builds

### PR DESCRIPTION
**Pull request by Neumann-A**

### CODE REVIEW

**Comment by JackBoosY** [comment]
```
Building package getopt-win32[core]:x64-windows-static...
-- Note: getopt-win32 only supports dynamic library linkage. Building dynamic library.
CMake Error at scripts/cmake/vcpkg_check_linkage.cmake:47 (message):
  Refusing to build unexpected dynamic library against the static CRT.  If
  this is desired, please configure your triplet to directly request this
  configuration.
Call Stack (most recent call first):
  ports/getopt-win32/portfile.cmake:5 (vcpkg_check_linkage)
  scripts/ports.cmake:142 (include)
```

File: ports/getopt-win32/portfile.cmake [hunk 1]
```diff
@@ -12,13 +10,39 @@ vcpkg_from_github(
...
 
 # Copy header
-file(COPY ${SOURCE_PATH}/getopt.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/)
+file(COPY "${SOURCE_PATH}/getopt.h" DESTINATION "${CURRENT_PACKAGES_DIR}/include/")
+set(_file "${CURRENT_PACKAGES_DIR}/include/getopt.h")
+file(READ "${_file}" _contents)
+if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
+    string(REPLACE "	#define __GETOPT_H_" "	#define __GETOPT_H_\n	#define STATIC_GETOPT" _contents "${_contents}")
+endif()
+file(WRITE "${_file}" "${_contents}")
```
Comment (JackBoosY) [inline ports/getopt-win32/portfile.cmake:46]:
```suggestion
if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/getopt.h"
        "	#define __GETOPT_H_" "	#define __GETOPT_H_\n	#define STATIC_GETOPT"
    )
endif()
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]
I wish `ConfigurationType` and `RuntimeLibrary` can be set as a option for `vcpkg_install_msbuild` or `vcpkg_build_msbuild`.

**Comment by JackBoosY** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/17983.

**Review by JackBoosY — APPROVED** [review]

**Comment by Neumann-A** [comment]
@JackBoosY Why did you mark this as requires discussion?

**Comment by JackBoosY** [comment]
I hope my colleague will do further review.

**Pull request merged**